controller/model/asynchronous_eventer.go

```diff
@@ -0,0 +1,280 @@
+
+	return nil
+}
+
+func (a *AsynchronousEventerBbolt) run() {
+
+	for {
+		select {
+		case <-a.closeNotify:
+			a.running.Set(false)
+			return
+		case <-a.stopNotify:
+			a.running.Set(false)
+			return
+		case <-time.After(a.Interval):
+			a.process()
+		}
+	}
+}
+
+func (a *AsynchronousEventerBbolt) Stop() error {
+	if a.running.CompareAndSwap(true, false) {
+		close(a.stopNotify)
+		return nil
+	}
+
+	return errors.New("not started")
+}
+
+func (a *AsynchronousEventerBbolt) Trigger() error {
+	if a.running.Get() {
+		go a.process()
+		return nil
+	}
+
+	return errors.New("not started")
+}
+
+func (a *AsynchronousEventerBbolt) deleteEvent(id string) error {
+	return a.env.GetDbProvider().GetDb().Update(func(tx *bbolt.Tx) error {
+		ctx := boltz.NewMutateContext(tx)
+		return a.env.GetStores().AsynchronousEvent.DeleteById(ctx, id)
+	})
+}
+
+func (a *AsynchronousEventerBbolt) process() {
+	if !a.processing.CompareAndSwap(false, true) {
+		return //already running
+	}
+
+	defer func() {
+		if r := recover(); r != nil {
+			var err error
+			switch x := r.(type) {
+			case string:
+				err = errors.New(x)
+			case error:
+				err = x
+			default:
+				err = errors.New("unknown panic")
+			}
+
+			pfxlog.Logger().WithError(err).Error("panic caught during delete processing")
+		}
+		a.processing.Set(false)
+	}()
+
+	for {
+		var count int64 = 0
+		var events []*persistence.AsynchronousEvent
+		err := a.env.GetDbProvider().GetDb().View(func(tx *bbolt.Tx) error {
+			var ids []string
+			var err error
+			ids, count, err = a.env.GetStores().AsynchronousEvent.QueryIds(tx, "limit 50")
+
+			if err != nil {
+				return err
+			}
+
+			for _, id := range ids {
+				event, err := a.env.GetStores().AsynchronousEvent.LoadOneById(tx, id)
+
+				if err != nil {
+					return fmt.Errorf("error could not load event id %s", id)
+				} else {
+					events = append(events, event)
+				}
+			}
+			return nil
+		})
+
+		if err != nil {
+			pfxlog.Logger().WithError(err).Error("error during AsynchronousEventerBbolt processing")
+
+		}
+
+		if count != int64(len(events)) {
+			pfxlog.Logger().Warnf("%d event records were queried and only %d events were loaded", count, len(events))
+		}
+
+		if len(events) == 0 {
+			return
+		}
+
+		for _, event := range events {
+			if handlerVals, found := a.handlerMap.Get(event.Type); found {
+				handlers := handlerVals.([]EventListenerFunc)
+				for _, handler := range handlers {
+					func() {
+						defer func() {
+							if r := recover(); r != nil {
+								var err error
+								switch x := r.(type) {
+								case string:
+									err = errors.New(x)
+								case error:
+									err = x
+								default:
+									err = errors.New("unknown panic")
+								}
+
+								pfxlog.Logger().WithError(err).Error("panic caught during asynchronous event %s handler %v", event.Type, handler)
+							}
+						}()
+
+						handler(event.Type, event.Data)
+					}()
+				}
+
+				if err := a.deleteEvent(event.Id); err != nil {
```

Comment:
Been trying to move any variables in logging into fields, for ease of parsing

Reply:
Got it. Fixed